File: 1.0/src/cz/mg/vulkan/PFNvkGetPhysicalDeviceExternalBufferPropertiesKHR.java
package cz.mg.vulkan;

public class PFNvkGetPhysicalDeviceExternalBufferPropertiesKHR extends VkFunctionPointer {
    public PFNvkGetPhysicalDeviceExternalBufferPropertiesKHR() {
    }

    protected PFNvkGetPhysicalDeviceExternalBufferPropertiesKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetPhysicalDeviceExternalBufferPropertiesKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetPhysicalDeviceExternalBufferPropertiesKHR(long value) {
        setValue(value);
    }

    public PFNvkGetPhysicalDeviceExternalBufferPropertiesKHR(VkInstance instance) {
        super(instance, new VkString("vkGetPhysicalDeviceExternalBufferPropertiesKHR"));
    }

    public void call(VkPhysicalDevice physicalDevice, VkPhysicalDeviceExternalBufferInfo pExternalBufferInfo, VkExternalBufferProperties pExternalBufferProperties){
        callNative(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.getNullAddressNative(), pExternalBufferInfo != null ? pExternalBufferInfo.getVkAddress() : VkPointer.NULL, pExternalBufferProperties != null ? pExternalBufferProperties.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long physicalDevice, long pExternalBufferInfo, long pExternalBufferProperties);



}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetPhysicalDeviceExternalBufferPropertiesKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetPhysicalDeviceExternalBufferPropertiesKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jlong pExternalBufferInfo, jlong pExternalBufferProperties){
    (void)env;
    (void)jc;
    PFN_vkGetPhysicalDeviceExternalBufferPropertiesKHR f = (PFN_vkGetPhysicalDeviceExternalBufferPropertiesKHR)jniLongToFunctionPointer(address);
    f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        ((VkPhysicalDeviceExternalBufferInfo*)jniLongToPointer(pExternalBufferInfo)),
        ((VkExternalBufferProperties*)jniLongToPointer(pExternalBufferProperties))
    );
}

